### PR TITLE
[TASK] Add `-b` option to select container binary

### PR DESCRIPTION
**Pull request by sbuerk**
TYPO3 core introduced the `-b` option for the
`Build/Scripts/runTests.sh` script to provide
the ability to select the container binary,
`docker` or `podman` and enforces podman for
CI (GitLab runner) runs after switching the
Core CI infrastructure from Docker-In-Docker
to Podman-In-Podman.

The `ext:tea` adoption of the `runTests.sh`
has been already adopted on a intermediate
version and dropped the `docker-compose`
based container definitions. Along with
other minor prepration patches this allows
now to add the container binary selection
also.

This change modifies `runTests.sh` to

* add the `-b` option with a corresponding
  helptext addition and allows `podman` or
  `docker` - whereas the latter is the default.

* integrates the determined and fixed issues
  with tty in CI environments by setting some
  paramater for the commands based on the CI
  detection (GitHub Action or GitLab Runner).

Note:	The implemention for `ext:tea` does not
        enforce hardcoed the `podman` binary for
        CI environments. If no user selection is
        provided, it fallbacks to `docker` for
        all environments.

It's worth to mention that the implementation to
deal properly with user namespacing and permission
handling on mountend volumns it's required to have
at least `podman 4.x` installed. Earlier versions
does not support the required handlinging and options.

Minor moving of variable declaration are done as a
drive-by change.

The `docsGenerate` suite has been guard and display
a concrete message when executed using the `podman`
binary.

Resolves: #1075
Related: #969

### CODE REVIEW

File: Build/Scripts/runTests.sh [hunk 1]
```diff
@@ -480,35 +518,44 @@ case ${TEST_SUITE} in
         ;;
     docsGenerate)
         # @todo contact the documentation team for a future rootles podman version
-        ${CONTAINER_BIN} run --rm ${IMAGE_DOCS} show-shell-commands > generate-documentation.sh
-        echo 'dockrun_t3rd makehtml' >> generate-documentation.sh
-        bash generate-documentation.sh
-        rm -Rf generate-documentation.sh
-        SUITE_EXIT_CODE=$?
+        if [ ${CONTAINER_BIN} = "podman" ]; the
```
Comment (simonschaufi) [inline Build/Scripts/runTests.sh:521]: then

File: Build/Scripts/runTests.sh [hunk 2]
```diff
@@ -362,41 +393,45 @@ done
...
 fi
 
-# Go to the directory this script is located, so everything else is relative
-# to this dir, no matter from where this script is called, then go up two dirs.
-THIS_SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
-cd "$THIS_SCRIPT_DIR" || exit 1
-cd ../../ || exit 1
-ROOT_DIR="${PWD}"
+if ! type ${CONTAINER_BIN} >/dev/null 2>&1; then
+    echo "Selected container environment \"${CONTAINER_BIN}\" not found. Please install or use -b option to select one." >&2
```
Comment (simonschaufi) [inline Build/Scripts/runTests.sh:427]: Please install what?

**Review by oliverklee — CHANGES_REQUESTED** [review]
This breaks seemingly all `runTests.sh` commands for me due to a syntax error:

```
klee@gonzales ~/src/typo3/ext/tea (podman)$ ./Build/Scripts/runTests.sh -s help
./Build/Scripts/runTests.sh: Zeile 417: [: =: Einstelliger (unärer) Operator erwartet.
./Build/Scripts/runTests.sh: Zeile 421: [: =: Einstelliger (unärer) Operator erwartet.
./Build/Scripts/runTests.sh: Zeile 446: network: Befehl nicht gefunden
./Build/Scripts/runTests.sh: Zeile 448: [: =: Einstelliger (unärer) Operator erwartet.
./Build/Scripts/runTests.sh: Zeile 529: Syntaxfehler beim unerwarteten Wort »else«
./Build/Scripts/runTests.sh: Zeile 529: `        else'
```

File: Build/Scripts/runTests.sh (hunk 1 above)
Comment (sbuerk) [reply Build/Scripts/runTests.sh]: fixed

File: Build/Scripts/runTests.sh (hunk 2 above)
Comment (sbuerk) [reply Build/Scripts/runTests.sh]: changed wording

**Comment by sbuerk** [comment]
> This breaks seemingly all `runTests.sh` commands for me due to a syntax error:
> 
> ```
> klee@gonzales ~/src/typo3/ext/tea (podman)$ ./Build/Scripts/runTests.sh -s help
> ./Build/Scripts/runTests.sh: Zeile 417: [: =: Einstelliger (unärer) Operator erwartet.
> ./Build/Scripts/runTests.sh: Zeile 421: [: =: Einstelliger (unärer) Operator erwartet.
> ./Build/Scripts/runTests.sh: Zeile 446: network: Befehl nicht gefunden
> ./Build/Scripts/runTests.sh: Zeile 448: [: =: Einstelliger (unärer) Operator erwartet.
> ./Build/Scripts/runTests.sh: Zeile 529: Syntaxfehler beim unerwarteten Wort »else«
> ./Build/Scripts/runTests.sh: Zeile 529: `        else'
> ```

Changed to a more safer variant. I'm curios - that's simply copied from the TYPO3 core runTests.sh - have you issues there too ?

However, should be fixed for here now.

**Comment by oliverklee** [comment]
No, `./Build/Scripts/runTests.sh -s help` on Core `main` strangely works for me without problems.

**Review by oliverklee — APPROVED** [review]

**Review by simonschaufi — APPROVED** [review]

**Pull request merged**